uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java

```diff
@@ -1,35 +1,209 @@
 package com.learning.uwuno.unit;
 
 import com.learning.uwuno.controllers.roomController;
-import com.learning.uwuno.controllers.playerController;
-
-
+import com.learning.uwuno.player;
+import com.learning.uwuno.room;
 import com.learning.uwuno.services.gameService;
+
 import org.junit.jupiter.api.Test;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
 import org.springframework.test.web.servlet.MockMvc;
 
+import static org.hamcrest.Matchers.hasSize;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
-import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.*;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 @WebMvcTest(roomController.class)
 @AutoConfigureMockMvc
 public class roomControllerTests {
     @Autowired
     private MockMvc mock;
 
+    @Autowired
+    private roomController controller;
+
     @MockBean
     private gameService service;
 
+    room room = new room("testRoom", false);
 
     @Test
-    public void testRoom_POST() throws Exception {
-//        this.mock.perform(post("/Rooms")).andDo(print()).andExpect(status().isOk());
+    public void GET_empty_roomList() throws Exception {
+        ArrayList<room> roomList = new ArrayList<>();
+        when(service.getRoomList()).thenReturn(roomList);
+
+        mock.perform(get("/rooms")
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().json("[]"))
+                .andDo(print());
+    }
+
+    @Test
+    public void GET_roomList() throws Exception {
+        ArrayList<room> roomList = new ArrayList<>();
+        roomList.add(room);
+        when(service.getRoomList()).thenReturn(roomList);
+
+        mock.perform(get("/rooms")
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0]['name']").value("testRoom"))
+                .andExpect(jsonPath("$.length()").value(1))
+                .andDo(print());
+    }
+
+    @Test
+    public void GET_room() throws Exception {
+        room.addPlayer(new player("testPlayer"));
```

Comment:
Maybe just create one instance of this right after room at the very beginning instead.